Description: Don't build with -O2.
 With gcc-4.6 4.6.1-15 on Debian unstable i386-kfreebsd something in -O2
 that can't be controlled by a flag causes problems with the bsnes GUI.
 The bsnes GUI doesn't show up at all, snespurify-gtk crashes all the time.
 Using -O1 and enabling all additional flags from -O3 works fine.
Origin: http://anonscm.debian.org/gitweb/?p=pkg-games/bsnes.git;a=tree;f=debian/patches

--- a/bsnes/Makefile
+++ b/bsnes/Makefile
@@ -11,7 +11,16 @@
 # compiler
 c       := $(compiler) -std=gnu99
 cpp     := $(subst cc,++,$(compiler)) -std=gnu++0x
-flags   := -I. -O3 -fomit-frame-pointer
+flags   := -I. -O1 -fthread-jumps -falign-functions  -falign-jumps -falign-loops \
+-falign-labels -fcaller-saves -fcrossjumping -fcse-follow-jumps \
+-fcse-skip-blocks -fdelete-null-pointer-checks -fdevirtualize \
+-fexpensive-optimizations -fgcse -fgcse-lm -finline-small-functions \
+-findirect-inlining -fipa-sra -foptimize-sibling-calls -fpartial-inlining \
+-fpeephole2 -fregmove -freorder-blocks -freorder-functions \
+-frerun-cse-after-loop -fsched-interblock -fsched-spec -fschedule-insns \
+-fschedule-insns2 -fstrict-aliasing -fstrict-overflow -ftree-switch-conversion \
+-ftree-pre -ftree-vrp -finline-functions -funswitch-loops -fpredictive-commoning \
+-fgcse-after-reload -fipa-cp-clone -fomit-frame-pointer
 link    :=
 objects := libco
 
--- a/snespurify/cc-gtk.sh
+++ b/snespurify/cc-gtk.sh
@@ -1,4 +1,23 @@
-g++ -std=gnu++0x -I. -O3 -fomit-frame-pointer -c phoenix/phoenix.cpp `pkg-config --cflags gtk+-2.0` -DPHOENIX_GTK
-g++ -std=gnu++0x -I. -O3 -fomit-frame-pointer -c snespurify.cpp
+g++ -std=gnu++0x -I. -O1 -fthread-jumps -falign-functions  -falign-jumps -falign-loops \
+-falign-labels -fcaller-saves -fcrossjumping -fcse-follow-jumps \
+-fcse-skip-blocks -fdelete-null-pointer-checks -fdevirtualize \
+-fexpensive-optimizations -fgcse -fgcse-lm -finline-small-functions \
+-findirect-inlining -fipa-sra -foptimize-sibling-calls -fpartial-inlining \
+-fpeephole2 -fregmove -freorder-blocks -freorder-functions \
+-frerun-cse-after-loop -fsched-interblock -fsched-spec -fschedule-insns \
+-fschedule-insns2 -fstrict-aliasing -fstrict-overflow -ftree-switch-conversion \
+-ftree-pre -ftree-vrp -finline-functions -funswitch-loops -fpredictive-commoning \
+-fgcse-after-reload -fipa-cp-clone -fomit-frame-pointer -c phoenix/phoenix.cpp \
+`pkg-config --cflags gtk+-2.0` -DPHOENIX_GTK
+g++ -std=gnu++0x -I. -O1 -fthread-jumps -falign-functions  -falign-jumps -falign-loops \
+-falign-labels -fcaller-saves -fcrossjumping -fcse-follow-jumps \
+-fcse-skip-blocks -fdelete-null-pointer-checks -fdevirtualize \
+-fexpensive-optimizations -fgcse -fgcse-lm -finline-small-functions \
+-findirect-inlining -fipa-sra -foptimize-sibling-calls -fpartial-inlining \
+-fpeephole2 -fregmove -freorder-blocks -freorder-functions \
+-frerun-cse-after-loop -fsched-interblock -fsched-spec -fschedule-insns \
+-fschedule-insns2 -fstrict-aliasing -fstrict-overflow -ftree-switch-conversion \
+-ftree-pre -ftree-vrp -finline-functions -funswitch-loops -fpredictive-commoning \
+-fgcse-after-reload -fipa-cp-clone -fomit-frame-pointer -c snespurify.cpp
 g++ -s -o snespurify-gtk snespurify.o phoenix.o `pkg-config --libs gtk+-2.0` -lX11
 rm *.o
